packages/kit/src/template.ts

```diff
@@ -113,18 +113,55 @@ export async function updateTemplates (options?: { filter?: (template: ResolvedN
 }
 
 export async function _generateTypes (nuxt: Nuxt) {
-  const nodeModulePaths = getModulePaths(nuxt.options.modulesDir)
-
   const rootDirWithSlash = withTrailingSlash(nuxt.options.rootDir)
+  const relativeRootDir = relativeWithDot(nuxt.options.buildDir, nuxt.options.rootDir)
 
-  const modulePaths = await resolveNuxtModule(rootDirWithSlash,
-    nuxt.options._installedModules
-      .filter(m => m.entryPath)
-      .map(m => getDirectory(m.entryPath!)),
-  )
+  const include = new Set<string>([
+    './nuxt.d.ts',
+    join(relativeRootDir, '.config/nuxt.*'),
+    join(relativeRootDir, '**/*'),
+  ])
 
-  const isV4 = nuxt.options.future?.compatibilityVersion === 4
+  if (nuxt.options.srcDir !== nuxt.options.rootDir) {
```

Comment:
Just wanted to say good call on this one, I decided to not refsctor these 1 element spread operators yet because I was working on stuff that has more impact as we discusses, and then forgot to fix that after I was done. 👍